coref/config.py

```diff
@@ -48,24 +52,43 @@ class Config:  # pylint: disable=too-many-instance-attributes, too-few-public-me
     bce_loss_weight: float
 
     tokenizer_kwargs: Dict[str, dict]
+
+    model_bank: ModelConfig = field(init=False)
+
     conll_log_dir: str
 
     def __post_init__(self):
-        with open(Path(self.train_data), "r") as f:
+        with open(self.train_data, "r") as f:
             self.num_of_training_docs = sum(1 for _ in f)
+        encoder, tokenizer = bert.load_bert(self)
+        self.model_bank =  ModelConfig(encoder, tokenizer)
+
 
     @staticmethod
     def load_config(config_path: str, section: str = "roberta") -> "Config":
         config = toml.load(config_path)
         default_section = config["DEFAULT"]
         current_section = config[section]
         unknown_keys = set(current_section.keys()) - set(default_section.keys())
+        updated_path_type = strs_2_paths(default_section, current_section)
         if unknown_keys:
             raise ValueError(f"Unexpected config keys: {unknown_keys}")
-        return Config(section, **{**default_section, **current_section})
+        return Config(
+            section, **{**default_section, **current_section, **updated_path_type}
+        )
 
     @staticmethod
     def load_default_config(section: Optional[str]) -> "Config":
         if section is not None:
             return Config.load_config("config.toml", section)
         return Config.load_config("config.toml")
+
+
+def strs_2_paths(
```

Comment:
This could be generalized as a utility function that will handle str paths, try to resolve them and raise errors beforehand. Adding it as a suggestion for future development.

Reply:
This is changes in the config PR :)